content/blog/cloud-native-secret-management-with-pulumi-esc-and-external-secrets-operator/index.md

```diff
@@ -0,0 +1,419 @@
+
+- `PushSecret`: With the PushSecret resource, you can push secrets from the Kubernetes cluster to the external secret management system. The Pulumi ESC provider for ESO supports this feature, which allows you to push secrets from the cluster into Pulumi ESC. This feature is useful when you create secrets through other means and want to synchronize them into Pulumi ESC to manage them centrally.
+
+## How to Use Pulumi ESC and ESO Together
+
+To demonstrate how to use Pulumi ESC and ESO together, we will store some secrets and configuration values in Pulumi ESC, then synchronize them into a Kubernetes cluster using ESO.
+
+To follow along, you'll need the following prerequisites:
+
+- A Kubernetes cluster (I will be using a local [KinD](https://kind.sigs.k8s.io/) cluster, but you can use any
+  Kubernetes cluster)
+- [Pulumi CLI](/docs/iac/download-install/) installed
+- `kubectl` [CLI installed](https://kubernetes.io/docs/tasks/tools/) for some debugging
+
+### Step 1: Install ESO and Link to Pulumi ESC
+
+We will use Pulumi with the `pulumi-kubernetes` provider to deploy ESO into the Kubernetes cluster. The power of Pulumi ESC integration comes in two ways: First, to supply the Pulumi access token to ESO via Pulumi IaC, allowing the new ESO instance to access Pulumi ESC later. Second, to provide configuration and secrets from Pulumi ESC, via ESO, to the `PodInfo` application within the Kubernetes cluster.
+
+```bash
+pulumi new kubernetes-typescript --name eso-pulumi-esc
+```
+
+You should see the following output:
+
+```bash
+This command will walk you through creating a new Pulumi project.
+
+Enter a value or leave blank to accept the (default), and press <ENTER>.
+Press ^C at any time to quit.
+
+Project description (A minimal Kubernetes TypeScript Pulumi program):
+Created project 'eso-pulumi-esc'
+
+Please enter your desired stack name.
+To create a stack in an organization, use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).
+Stack name (dev):
+Created stack 'dev'
+
+The package manager to use for installing dependencies npm
+Installing dependencies...
+
+
+added 310 packages, and audited 311 packages in 12s
+
+44 packages are looking for funding
+  run `npm fund` for details
+
+found 0 vulnerabilities
+Finished installing dependencies
+
+Your new project is ready to go! ✨
+
+To perform an initial deployment, run `pulumi up`
+```
+
+Head to Pulumi Cloud Console and create a new Pulumi ESC project with the name `eso-esc-demo`:
+
+![Pulumi ESC Environment](env1.png)
+
+And the environment `eso-esc-dev`:
+
+![Pulumi ESC Environment](env2.png)
+
+In the editor add the following yaml into the `Environment definition`:
+
+```yaml
+values:
+  pulumiConfig:
+    pulumi-pat:
+      fn::secret: <your-pulumi-pat>
+```
+
+{{% notes type="info" %}}
```

Comment:
I could expand this to actually say "Pulumi access token" and not keep it as a note